services/common/serv_helper.js

```diff
@@ -40,6 +40,21 @@
     });
   };
 
+  var cloneObject = function(obj) {
```

Comment:
Going to merge this, but add a comment here saying that circular objects will cause this to hang. Can always fix this if/when run into this problem.